### PR TITLE
gear verifier & eth-client

### PR DESCRIPTION
**Pull request by LouiseMedova**

### CODE REVIEW

File: solidity_verifier/hardhat/test/verify.ts
Comment (mertwole) [inline solidity_verifier/hardhat/test/verify.ts:1]: Do you need this file?

File: eth-client/Cargo.toml
Comment (mertwole) [inline eth-client/Cargo.toml:1]: Please derive some non ethereum-specific deps from workspace

File: eth-client/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    pub async fn verify_vs_change(
+        &self,
+        p_a: [U256; 2],
+        p_b: [[U256; 2]; 2],
+        p_c: [U256; 2],
+        validator_set: [U256; 5],
+        nonce_id: U256,
+    ) -> Result<bool, VerifierError> {
+        let account = {
+            let pk = env::var("PK").expect("PK is not set");
```
Comment (mertwole) [inline eth-client/src/lib.rs:45]: I think it's better to turn it into the parameter to get it from args/env in relayer

File: eth-client/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            .parse::<web3::types::Address>()
+            .map_err(|_| VerifierError::WrongAddress)?;
+
+        Ok(ContractVerifiers {
+            vs_change_vrf: ValidatorSetChangeVerifier::at(&web3, address_vs_change),
+            msg_sent_vrf: MessageSentVerifier::at(&web3, address_msg_sent),
+        })
+    }
+
+    pub async fn verify_vs_change(
```
[the remark this answers is not in the document]
Comment (LouiseMedova) [reply eth-client/src/lib.rs:36]: ok

File: solidity_verifier/hardhat/test/verify.ts
Comment (LouiseMedova) [reply solidity_verifier/hardhat/test/verify.ts:1]: No

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,7 @@
 target/
-.vscode
\ No newline at end of file
+.vscode
+node_modules
+typechain-types
+artifacts
+cache
+.DS_Store
```
Comment (mertwole) [inline .gitignore:7]: Please remove these files from git too

File: eth-client/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+tokio = { version ="1.35.1",  features = ["full"] }
+ethcontract = "0.25.4"
+serde = "1.0.195"
+serde_json = "1.0.111"
+primitive-types = "0.12.2"
+web3 = { version = "0.19", default-features = false, features = ["signing"] }
```
Comment (mertwole) [inline eth-client/Cargo.toml:12]: Please put all the deps except `ethcontract` and `web3` inside root `Cargo.toml`

File: eth-client/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,325 @@
+mod error;
+use error::VerifierError;
+use ethcontract::prelude::*;
+use primitive_types::U256;
+use serde::Deserialize;
+use std::{env, fs};
+ethcontract::contract!("solidity_verifier/hardhat/artifacts/contracts/vs_change/validator_change_verifier.sol/ValidatorSetChangeVerifier.json");
```
Comment (mertwole) [inline eth-client/src/lib.rs:7]:
```suggestion
use std::{env, fs};

ethcontract::contract!("solidity_verifier/hardhat/artifacts/contracts/vs_change/validator_change_verifier.sol/ValidatorSetChangeVerifier.json");
```

File: eth-client/src/lib.rs
Comment (mertwole) [inline eth-client/src/lib.rs:1]: clippy

File: .gitignore (hunk 3 above)
Comment (mertwole) [reply .gitignore:7]: and also it seems like you need to ignore these folders like for example `**/node_modules`

File: eth-client/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+            vs_change_vrf: ValidatorSetChangeVerifier::at(&web3, address_vs_change),
+            msg_sent_vrf: MessageSentVerifier::at(&web3, address_msg_sent),
+        })
+    }
+
+    pub async fn verify_vs_change(
+        &self,
+        pk: String,
+        path_to_final_proof: &str,
+        path_to_final_public: &str,
```
Comment (mertwole) [inline eth-client/src/lib.rs:41]: `PathBuf` here

File: eth-client/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+            return Ok(false);
+        }
+        Ok(true)
+    }
+
+    pub async fn verify_msg_sent(
+        &self,
+        pk: String,
+        path_to_final_proof: &str,
+        path_to_final_public: &str,
```
Comment (mertwole) [inline eth-client/src/lib.rs:84]: `PathBuf` here

**Pull request merged**